plugin-server/src/worker/ingestion/event-pipeline/processAiEvent.ts

```diff
@@ -0,0 +1,50 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+import bigDecimal from 'js-big-decimal'
+
+import { ModelRow } from '../../../types'
+import { providers } from '../../../utils/ai-cost-data/mappings'
+
+export const processAiEvent = (event: PluginEvent): PluginEvent => {
```

Comment:
I fixed up the bit I had issue. Main thing now is tests. Ingestion is hyper critical so tests are a must especially for this kind of code that seems to do a lot of loops...

Reply:
added